charts/hivemq-platform/templates/hivemq-custom-resource.yml

```diff
@@ -141,15 +142,23 @@ spec:
               image: "{{ .Values.image.repository }}/{{ .Values.image.name }}:{{ .Values.image.tag }}"
               imagePullPolicy: {{ .Values.image.pullPolicy }}
               ports:
+                {{- $containerPortNameList := list }}
+                {{- $containerPortName := "" }}
                 {{- range $key, $val := .Values.services }}
-                {{- if $val.exposed}}
-                - name: {{ include "hivemq-platform.range-service-port-name" . | trimAll "-" | trunc 15 | trimSuffix "-" |  trim }}
-                  containerPort: {{$val.containerPort}}
+                  {{- $containerPortName = (include "hivemq-platform.range-service-port-name" .) | trimAll "-" | trunc 15 | trimSuffix "-" | trim }}
```

Comment:
Can or should we move the `trimAll "-" | trunc 15 | trimSuffix "-" | trim` to the helper method, like we it for `hivemq-platform.range-service-name`? That would make the usage less error prone.

I'm also not sure if we have a check somewhere if the truncation leads to duplicate names. Not sure if this can happen for port names at all, since they are not customizable and we control all the prefixes, right?

Reply:
As explained, problem here is that this function is used in two different places with different truncate values. So I could pass this truncate size as a parameter to the helper function but doing that I will need to define as well all the different parameters already used by this function cause doing this, the context is lost (for instance, the .Values are not accessible anymore).

But I think it would be better to have two different functions, even with the duplicated logic (it's not that big), one for the service port name (63 chars max) and another one for the container port name (15 chars max). Will do that.